fixing #50

Lucas-Fernandes-Martins opened the pull request:
# Pull Request

## Description

Please share details of what you have added and why?

Adding rolling_summary option as discussed [here](https://github.com/meta-llama/synthetic-data-kit/issues/50).

Fixes # (issue)

#50 

## Type of change
Bug fix

Please non-releavant options

- [ check] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change
- [ ] Documentation update

init27 commented on synthetic_data_kit/generators/qa_generator.py:
```diff
@@ -40,15 +42,39 @@ def generate_summary(self, document_text: str) -> str:
+            #split text into long chunks for summarization
+            chunks = split_into_chunks(document_text,
+                                       chunk_size=max_context_length,
+                                       overlap=0)
```
Open Q-Should this also be exposed via the config?

If not, is it worth having it as a variable here?

Lucas-Fernandes-Martins replied:
Hi, thanks for reviewing it! 

Right now, I'm using max_content_length, which is already defined in the configs/config.yaml.

I just wanted to ask why max_content_length's default value is currently 8k. I guess for most models we would be able to go significantly higher.

Doing some experiments I got to the conclusion that 100k chars seems like a good balance between summary being generated fast and not risking going over the context window of most LLM's). Most models nowadays seem to have at least 128k tokens of context, so 100k chars seemed like a good conservative choice. However, using 8k will make summary generation very slow.

Let me know what you think and if I should create a different config variable than max_content_length specifically to represent the summary chunks.

Thank you :)

init27 replied:
Yes this is great!

init27 commented:
@Lucas-Fernandes-Martins could you please push a small test for the CI/CD test that is breaking since we added the param?

Lucas-Fernandes-Martins commented:
Hi @init27 sorry, it was failing because I forgot to add rolling_summary as Optional parameter so it was giving a Type error. I fixed it and corrected the PR, so now it should be go to go.

Let me know if you'd like me to write a specific unit test for the rolling_summary flag or change an existing test.

Thanks!
